sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/connection/MSSQLConnectionProvider.scala

```diff
@@ -38,18 +38,16 @@ private[sql] class MSSQLConnectionProvider extends SecureConnectionProvider {
     val parseURL = try {
       // The default parser method signature is the following:
       // private Properties parseAndMergeProperties(String Url, Properties suppliedProperties)
-      val m = driver.getClass.getDeclaredMethod(parserMethod, classOf[String], classOf[Properties])
-      m.setAccessible(true)
```

Comment:
This looks irrelevant from the PR title. Why do we remove `setAccessible` here?

`parseAndMergeProperties` is `private`, isn't it? If this is not `private`, we need to change the comment at line 40 together.